### ISSUE TITLE
GeoPartial Queries

### ISSUE THREAD

**Issue by arunoda** [arunoda/meteor-smart-collections#23]
Since, minimongo query selector does not supports [geo-partial](http://docs.mongodb.org/manual/reference/operator/query-geospatial/) queries this SmartCollections lacks it.

**Comment by arunoda** [comment]
related to https://github.com/meteor/meteor/pull/443

**Comment by arunoda** [comment]
For temporarily, I will move SmartCollection back to re-run cursor for write operations when it detects a GeoPartial Query.

**Comment by iplanwebsites** [comment]
Do SmartCollections supports GeoSpatial Querries (ie. $near) on the client-side?
What are the limitation with re-run? The dataset isn't reactive?
Please let me know, I'm dying to implement some module that requires spatial selectors.
Best,

**Comment by arunoda** [comment]
Yes. Geo partial is not supported at this moment. It is not available on
the client even if you use meteor collections.

You have to use normal collections if you need geo partial support.

On Monday, September 9, 2013, I Plan Websites wrote:

> Do SmartCollections supports GeoSpatial Querries (ie. $near) on the
> client-side?
> What are the limitation with re-run? The dataset isn't reactive?
> Please let me know, I'm dying to implement some module that requires
> spatial selectors.
> Best,
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/arunoda/meteor-smart-collections/issues/23#issuecomment-24075806
> .

## 

Arunoda Susiripala

@arunoda http://twitter.com/arunoda
http://gplus.to/arunodahttps://github.com/arunoda
http://www.linkedin.com/in/arunoda

**Comment by iplanwebsites** [comment]
Thanks for the answer!
I'll stick to the standard Meteor Collection then and will wait until they finally implement $near selector. Cheers,

**Comment by arunoda** [comment]
Yep. Thats the solution for now. I can add geo query support but there
won't be any improvement. So better to wait.

On Monday, September 9, 2013, I Plan Websites wrote:

> Thanks for the answer!
> I'll stick to the standard Meteor Collection then and will wait until they
> finally implement $near selector. Cheers,
> 
> On 2013-09-09, at 10:38 AM, Arunoda Susiripala wrote:
> 
> > Yes. Geo partial is not supported at this moment. It is not available on
> > the client even if you use meteor collections.
> > 
> > You have to use normal collections if you need geo partial support.
> > 
> > On Monday, September 9, 2013, I Plan Websites wrote:
> > 
> > > Do SmartCollections supports GeoSpatial Querries (ie. $near) on the
> > > client-side?
> > > What are the limitation with re-run? The dataset isn't reactive?
> > > Please let me know, I'm dying to implement some module that requires
> > > spatial selectors.
> > > Best,
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub<
> > > https://github.com/arunoda/meteor-smart-collections/issues/23#issuecomment-24075806>
> > > 
> > > .
> > 
> > ## 
> > 
> > Arunoda Susiripala
> > 
> > @arunoda http://twitter.com/arunoda
> > http://gplus.to/arunodahttps://github.com/arunoda
> > http://www.linkedin.com/in/arunoda
> > —
> > Reply to this email directly or view it on GitHub.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/arunoda/meteor-smart-collections/issues/23#issuecomment-24086571
> .

## 

Arunoda Susiripala

@arunoda http://twitter.com/arunoda
http://gplus.to/arunodahttps://github.com/arunoda
http://www.linkedin.com/in/arunoda

**Comment by igalgh** [comment]
I noticed the $near queryop is now working. Is there some sort of a time table to get the other geopartial commands (for example: $geoWithin) working (?)

**Comment by arunoda** [comment]
I think that's because of mini-mongo(meteor's client mongo implementation)
 have it. I'm not sure when it will be available.

On Thu, Jan 2, 2014 at 5:15 AM, igaltivoni notifications@github.com wrote:

> I noticed the $near queryop is now working. Want to ask if there's some
> sort of a time table to get the other PeoPartial commands (for example:
> $geoWithin) working as well (?)
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/arunoda/meteor-smart-collections/issues/23#issuecomment-31433001
> .

## 

Arunoda Susiripala

@arunoda http://twitter.com/arunoda
http://gplus.to/arunodahttps://github.com/arunoda
http://www.linkedin.com/in/arunoda
